code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -91,6 +91,10 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 				possible_names += new type
 	namedatum = pick(possible_names)
 	updatetheme(pickedtheme)
+	
+/mob/living/simple_animal/hostile/guardian/get_spans()
```

Comment:
Errrrr you're returning null if it's NOT tech. Do this:
```
/mob/living/simple_animal/hostile/guardian/get_spans()
	. = ..()
	if(pickedtheme == tech)
		. |= SPAN_ROBOT
```

Reply:
ah, thanks!

Reply:
the picked theme is never saved anywhere arm, you need to add a new var to the mob
`current_theme` or similar

and then in the `updatetheme` proc set the `current_theme` var

Then you can reference it in this var to return the right spans.

Also it needs to be `if(pickedtheme == "tech")`

Reply:
Previously it's only been used to change the name/icon of the mob, so it's never been necessary to keep it around.

Reply:
or just add the span where the name/icon is set?

Reply:
is there a list/var that say code picks up ?

Reply:
Hmm, before tongues I think there was, probably not anymore good point.

Reply:
OK LETS DO THIS!